### PR TITLE
Add com.sun.jndi.url.dns to java9.options file

### LINKED ISSUE #25210: DnsContextFactory not accessible in java 17
**Describe the bug**  
With Java 17 and Liberty trying to make a dns lookup fails. A ffdc is created with this message:

```
Exception = java.lang.IllegalAccessException
Source = com.ibm.ws.jndi.internal.WASURLObjectFactoryFinder
probeid = 81
Stack Dump = java.lang.IllegalAccessException: class com.ibm.ws.jndi.internal.WASURLObjectFactoryFinder cannot access class com.sun.jndi.url.dns.dnsURLContextFactory (in module jdk.naming.dns) because module jdk.naming.dns does not export com.sun.jndi.url.dns to unnamed module @188cf1f7
```

**Steps to Reproduce**  
```
Properties env = new Properties();
env.put(Context.INITIAL_CONTEXT_FACTORY, "com.sun.jndi.dns.DnsContextFactory");
env.put(Context.PROVIDER_URL, "dns:");
dirContext = new InitialDirContext(env);
 
Attributes attrs = dirContext.getAttributes("dns:/...");
```

**Expected behavior**  
Successful dns lookup

**Diagnostic information:**  
 - OpenLiberty Version: 23.0.0.3
 - Java Version: Java 17

**Additional context**  
I think it is necessary to add this to the java9.options starting with java 17. This resolves my problem:

```
--add-exports
jdk.naming.dns/com.sun.jndi.url.dns=ALL-UNNAMED
```

https://github.com/OpenLiberty/open-liberty/blob/integration/dev/com.ibm.ws.kernel.boot/publish/platform/java/java9.options

### PR DESCRIPTION
**Pull request by gjwatts**
Fixes #25210

### CODE REVIEW

**Comment by gjwatts** [comment]
#build

**Comment by gjwatts** [comment]
#build

**Comment by gjwatts** [comment]
#build

**Comment by gjwatts** [comment]
#libbybot

File: dev/io.openliberty.java.internal_fat/bnd.bnd [hunk 1]
```diff
@@ -14,7 +14,8 @@
 bVersion=1.0
 
 src: \
-	fat/src
+	fat/src, \
+	test-applications/illegalaccess/src
```
Comment (jhanders34) [inline dev/io.openliberty.java.internal_fat/bnd.bnd:18]: copyright date update?

File: dev/io.openliberty.java.internal_fat/fat/src/io/openliberty/java/internal/fat/JavaIllegalAccessTest.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+ * Copyright (c) 2023 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
```
Comment (jhanders34) [inline dev/io.openliberty.java.internal_fat/fat/src/io/openliberty/java/internal/fat/JavaIllegalAccessTest.java]:
```suggestion
```

Not needed in new files.

File: dev/io.openliberty.java.internal_fat/publish/servers/java-illegal-access-exception-server/bootstrap.properties [hunk 3]
```diff
@@ -0,0 +1,14 @@
+###############################################################################
+# Copyright (c) 2017 IBM Corporation and others.
```
Comment (jhanders34) [inline dev/io.openliberty.java.internal_fat/publish/servers/java-illegal-access-exception-server/bootstrap.properties]: Is this file even needed?

File: dev/io.openliberty.java.internal_fat/publish/servers/java-illegal-access-exception-server/server.xml [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+    Copyright (c) 2023 IBM Corporation and others.
+    All rights reserved. This program and the accompanying materials
+    are made available under the terms of the Eclipse Public License 2.0
+    which accompanies this distribution, and is available at
+    http://www.eclipse.org/legal/epl-2.0/
+    
+    SPDX-License-Identifier: EPL-2.0
+   
+    Contributors:
+        IBM Corporation - initial API and implementation
```
Comment (jhanders34) [inline dev/io.openliberty.java.internal_fat/publish/servers/java-illegal-access-exception-server/server.xml]:
```suggestion
```

Not needed in new files.

File: dev/io.openliberty.java.internal_fat/publish/servers/java-illegal-access-exception-server/server.xml [hunk 5]
```diff
@@ -0,0 +1,26 @@
+<!--
+    Copyright (c) 2023 IBM Corporation and others.
+    All rights reserved. This program and the accompanying materials
+    are made available under the terms of the Eclipse Public License 2.0
+    which accompanies this distribution, and is available at
+    http://www.eclipse.org/legal/epl-2.0/
+
```
Comment (jhanders34) [inline dev/io.openliberty.java.internal_fat/publish/servers/java-illegal-access-exception-server/server.xml]:
```suggestion

```

File: dev/io.openliberty.java.internal_fat/test-applications/illegalaccess/src/test/web/IllegalAccessTestServlet.java [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+ * Copyright (c) 2023 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
```
Comment (jhanders34) [inline dev/io.openliberty.java.internal_fat/test-applications/illegalaccess/src/test/web/IllegalAccessTestServlet.java]:
```suggestion
```

not needed in new files.

**Comment by gjwatts** [comment]
#libbybot

**Comment by gjwatts** [comment]
#build

File: dev/io.openliberty.java.internal_fat/publish/servers/java-illegal-access-exception-server/server.xml [hunk 7]
```diff
@@ -0,0 +1,22 @@
+<!--
+    Copyright (c) 2023 IBM Corporation and others.
+    All rights reserved. This program and the accompanying materials
+    are made available under the terms of the Eclipse Public License 2.0
+    which accompanies this distribution, and is available at
+    http://www.eclipse.org/legal/epl-2.0/
+    SPDX-License-Identifier: EPL-2.0
```
Comment (jhanders34) [inline dev/io.openliberty.java.internal_fat/publish/servers/java-illegal-access-exception-server/server.xml]:
```suggestion
    http://www.eclipse.org/legal/epl-2.0/

    SPDX-License-Identifier: EPL-2.0
```

**Review by jhanders34 — DISMISSED** [review]

**Comment by gjwatts** [comment]
#build

**Comment by gjwatts** [comment]
#libbybot

**Comment by gjwatts** [comment]
#libbybot

**Comment by gjwatts** [comment]
#build

**Review by jhanders34 — APPROVED** [review]

**Pull request merged**